cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMirrorMaker2AssemblyOperator.java

```diff
@@ -0,0 +1,265 @@
+import io.strimzi.operator.common.operator.resource.ReconcileResult;
+import io.strimzi.operator.common.operator.resource.StatusUtils;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+
+/**
+ * <p>Assembly operator for a "Kafka Mirror Maker 2" assembly, which manages:</p>
+ * <ul>
+ *     <li>A Kafka Connect Deployment and related Services</li>
+ *     <li>A set of Mirror Maker 2 connectors</li>
+ * </ul>
+ */
+public class KafkaMirrorMaker2AssemblyOperator extends AbstractConnectOperator<KubernetesClient, KafkaMirrorMaker2, KafkaMirrorMaker2List, DoneableKafkaMirrorMaker2, Resource<KafkaMirrorMaker2, DoneableKafkaMirrorMaker2>> {
+
+    private static final Logger log = LogManager.getLogger(KafkaMirrorMaker2AssemblyOperator.class.getName());
+    private final DeploymentOperator deploymentOperations;
+    private final KafkaVersion.Lookup versions;
+
+    public static final String MIRRORMAKER2_SOURCE_CONNECTOR_SUFFIX = ".MirrorSourceConnector";
+    public static final String MIRRORMAKER2_CHECKPOINT_CONNECTOR_SUFFIX = ".MirrorCheckpointConnector";
+    public static final String MIRRORMAKER2_HEARTBEAT_CONNECTOR_SUFFIX = ".MirrorHeartbeatConnector";
+    public static final Map<String, Function<KafkaMirrorMaker2MirrorSpec, KafkaConnectorSpec>> MIRRORMAKER2_CONNECTORS = new HashMap<>();
+
+    /**
+     * @param vertx The Vertx instance
+     * @param pfa Platform features availability properties
+     * @param supplier Supplies the operators for different resources
+     * @param config ClusterOperator configuration. Used to get the user-configured image pull policy and the secrets.
+     */
+    public KafkaMirrorMaker2AssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa,
+                                        ResourceOperatorSupplier supplier,
+                                        ClusterOperatorConfig config) {
+        this(vertx, pfa, supplier, config, connect -> new KafkaConnectApiImpl(vertx));
+    }
+
+    public KafkaMirrorMaker2AssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa,
+                                        ResourceOperatorSupplier supplier,
+                                        ClusterOperatorConfig config,
+                                        Function<Vertx, KafkaConnectApi> connectClientProvider) {
+        super(vertx, pfa, KafkaMirrorMaker2.RESOURCE_KIND, supplier.mirrorMaker2Operator, supplier, config, connectClientProvider);
+        this.deploymentOperations = supplier.deploymentOperations;
+        this.versions = config.versions();
+
+        MIRRORMAKER2_CONNECTORS.put(MIRRORMAKER2_SOURCE_CONNECTOR_SUFFIX, mirror -> mirror.getSourceConnector());
+        MIRRORMAKER2_CONNECTORS.put(MIRRORMAKER2_CHECKPOINT_CONNECTOR_SUFFIX, mirror -> mirror.getCheckpointConnector());
+        MIRRORMAKER2_CONNECTORS.put(MIRRORMAKER2_HEARTBEAT_CONNECTOR_SUFFIX, mirror -> mirror.getHeartbeatConnector());
```

Comment:
Since `MIRRORMAKER2_CONNECTORS` is constant you should do this in a `static` init block immediately after the declaration of `MIRRORMAKER2_CONNECTORS`, rather than in this constructor. You should also be able to use a method reference instead of the lambda.

Reply:
Done